admin/admin-classic-editor.php

```
@@ -171,23 +171,18 @@ public function post_lang_choice() {
 			wp_die( 0 );
 		}
 
-		$post_type_object = get_post_type_object( $post_type );
-
-		if ( empty( $post_type_object ) ) {
-			wp_die( 0 );
-		}
-
-		if ( ! current_user_can( $post_type_object->cap->edit_post, $post_ID ) ) {
-			wp_die( -1 );
+		$return = $this->model->post->update_language( $post_ID, $lang, $post_type );
+		if ( is_wp_error( $return ) ) {
+			// phpcs:disable PSR2.ControlStructures.SwitchDeclaration.TerminatingComment
+			switch ( $return->get_error_code() ) {
+				case 'missing_capability':
+					wp_die( -1 );
+				default:
+					wp_die( 0 );
+			}
```

`wp_die()` should return an error message instead of an int.